packages/compiler/lib/src/encoder.dart

```diff
@@ -191,11 +194,18 @@ class PaintingCodec extends StandardMessageCodec {
   }
 
   void _writeDrawCommand(WriteBuffer buffer, DrawCommand command) {
-    buffer.putUint8(_drawCommandTag);
-    assert(_pathIds.containsKey(command.path), 'Expected to find ${command.path.hashCode}, have ${_pathIds.keys.map((p) => p.hashCode).toList()}');
-    assert(_paintIds.containsKey(command.paint));
-    buffer.putInt32(_pathIds[command.path]!);
-    buffer.putInt32(_paintIds[command.paint]!);
+    if (command is DrawPathCommand) {
```

Comment:
nit: we may want to have an enumeration of these as a separate field so we can be more confident about switch statements having all the cases, i.e. `enum DrawCommandType { path, vertices }`